### ISSUE TITLE
Error raised when deprecation warnings treated as errors

### ISSUE THREAD

**Issue by andrewjamesbrown** [chef-cookbooks/chef_handler#45]
### Cookbook version
1.4.0

### Chef-client version
12.16.42

### Platform Details
Ubuntu 14.04

### Scenario:
`chef-client` raises an exception when running with the `--config-option treat_deprecation_warnings_as_errors=true` option

### Steps to Reproduce:
Define new handler run *at compile time* and then run with `--config-option treat_deprecation_warnings_as_errors=true`

### Expected Result:
Cookbook converges successfully.

### Actual Result:
Chef-client run raises exception

Related issues:
https://github.com/chef/chef/pull/2853
https://github.com/chef-cookbooks/ark/issues/92
https://github.com/chef/chef/issues/2575

```
    NoMethodError
    -------------
    undefined method `new' for ChefHandler:Module

    Cookbook Trace:
    ---------------
    /var/chef/git/cookbooks-upstream/chef_handler/providers/default.rb:68:in `load_current_resource'
    /var/chef/git/cookbooks/base/recipes/failed-run-reporting.rb:15:in `from_file'

    Resource Declaration:
    ---------------------
    # In /var/chef/git/cookbooks/base/recipes/failed-run-reporting.rb

     11: chef_handler 'Chef::Handler::FailedRunHandler' do
     12:   source file
     13:   supports report: false, exception: true
     14:   action :nothing
     15: end.run_action(:enable)

    Compiled Resource:
    ------------------
    # Declared in /var/chef/git/cookbooks/base/recipes/failed-run-reporting.rb:11:in `from_file'

    chef_handler("Chef::Handler::FailedRunHandler") do
      action [:nothing]
      supports {:report=>false, :exception=>true}
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      declared_type :chef_handler
      cookbook_name :base
      recipe_name "failed-run-reporting"
      source "/var/chef/handlers/failed_run_handler.rb"
    end

    Platform:
    ---------
    x86_64-linux
```

Stacktrace:
```
NoMethodError: chef_handler[Chef::Handler::FailedRunHandler] (base::failed-run-reporting line 11) had an error: NoMethodError: undefined method `new' for ChefHandler:Module
/var/chef/git/cookbooks-upstream/chef_handler/providers/default.rb:68:in `load_current_resource'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/provider.rb:128:in `run_action'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/resource.rb:622:in `run_action'
/var/chef/git/cookbooks/base/recipes/failed-run-reporting.rb:15:in `from_file'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/mixin/from_file.rb:30:in `instance_eval'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/mixin/from_file.rb:30:in `from_file'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/cookbook_version.rb:233:in `load_recipe'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/run_context.rb:338:in `load_recipe'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/run_context/cookbook_compiler.rb:141:in `block in compile_recipes'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/run_context/cookbook_compiler.rb:138:in `each'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/run_context/cookbook_compiler.rb:138:in `compile_recipes'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/run_context/cookbook_compiler.rb:75:in `compile'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/run_context.rb:187:in `load'
/usr/local/chef-git/lib/chef_git/expand_node_object.rb:73:in `setup_run_context'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/client.rb:510:in `setup_run_context'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/client.rb:280:in `run'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/application.rb:276:in `run_with_graceful_exit_option'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/application.rb:252:in `block in run_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/local_mode.rb:44:in `with_server_connectivity'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/application.rb:235:in `run_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/application/client.rb:427:in `run_application'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/application.rb:59:in `run'
/opt/chef/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/bin/chef-client:26:in `<top (required)>'
/usr/local/bin/chef-git-client:8:in `load'
/usr/local/bin/chef-git-client:8:in `<main>'
```

**Comment by andrewjamesbrown** [comment]
@tas50 let me know how I can help here?

**Comment by tas50** [comment]
I think this is actually due to the use of supports on the resource. This was fixed in chef master and will go out in 13. I'm going to keep this issue open though until we can get a Chef 13 build to confirm

https://github.com/chef/chef/blob/master/RELEASE_NOTES.md#removed-supports-api-from-chefresource

**Comment by tas50** [comment]
This will be fixed by https://github.com/chef-cookbooks/chef_handler/pull/59

**Comment by tas50** [comment]
This is resolved in the 3.0 release
